### DISCUSSION TITLE
Alt + Mouse Not Working for Selection Subtraction in Krita

### DISCUSSION THREAD

**Discussion by Speyll** [labwc/labwc#2071] [Q&A] [answered]
First of all let me thank all of you guys working on Labwc since the implementation of tablet pen pressure, I've switched to Labwc as my daily driver and it's been a wonderful experience!

Now, while working in Krita, I encountered an issue with the selection tool. Normally, you can add to a selection with `Shift + Mouse` and remove from a selection with `Alt + Mouse`. However, when I try to use `Alt + Mouse` to subtract from the selection, instead of modifying the selection, it triggers the window movement action.

So what I did is I removed the keybinding for moving windows with `Alt + Mouse` in Labwc's configuration like this:

```
<mousebind button="A-Left" action="Drag">
	<action name="None" />
</mousebind>
```
                
This successfully stopped the Krita window from moving, but it did not resolve the issue with the selection tool in Krita. as you can see here:

https://github.com/user-attachments/assets/725c3d1d-ea56-40d2-95f7-4cc6f2d51c1d

If the video above doesn't work [here is a link](https://i.imgur.com/kZpHjss.mp4)

it still won't subtract selections.
A solution would be to rebind the binding in Krita but you see I'm the kind of guy that likes getting used to default Keybindings of the software i use.

Any guidance or suggestions would be greatly appreciated!

**Comment by droc12345** [comment]
Rather than try and set action to none why not modify the key binding
change A-Left to be S-A-Left or C-A-Left, something not in use by anything else.

**Reply by Speyll** [reply]
I tried that, sadly if I don't specifically set it to None (or leave it blank) it will default to moving the window, changing the keybind or even removing it completely still moves the window, I think it's a default behavior of Labwc unless overwritten.

**Reply by dashavoo** [reply, marked as answer]
I can confirm the same problem with selection subtraction in Krita (and selection moving in Gimp, which is the same keybind).
As you say, setting no action disables moving the window without letting the Krita binding work (I tried using an `If` action to remove the alt+left binding for krita and gimp only).
You probably have `<default />` in your mouse section. Removing that from my configuration and not re-adding a binding for alt+left allowed me to use the subtract-from-selection function in Krita, but losing the binding for labwc. If you want to have all default bindings except that one, I think you need to copy all the defaults into your own rc file.

**Reply by Speyll** [reply]
Thank you very much! that was it i didn't notice the `<default />` setting, commenting that out did the trick!

**Reply by Consolatis** [reply]
Sounds like `A-Left` for the `Press` action also needs to be disabled as well. See https://github.com/labwc/labwc/issues/2062#issuecomment-2276876266 for an example (instead of having the empty bindings you could also add the `None` action, it shouldn't make a difference).

In general I would recommend using `<default />` and then only adding / overwriting / removing bindings as necessary. But copying over everything also works.

**Reply by Speyll** [reply]
That did work indeed with `<default />` still enabled, very nice thank you!

**Comment by droc12345** [comment]
I usually copy rc.xml.all to my local rc.xml file and then edit or throw away anything that I don't need or want.
Changing the keybind should work, I override a lot of default settings.
